### PR TITLE
Add missing tests

### PR DESCRIPTION
**Pull request by martin-springer**
This is a draft. Mainly to get Codecov updated.

- [x] Code changes are covered by tests
- [x] Code changes have been evaluated for compatibility/integration with TrendAnalysis
- [x] New functions added to `__init__.py`
- [x] API.rst is up to date, along with other sphinx docs pages
- [x] Example notebooks are rerun and differences in results scrutinized
- [ ] Updated changelog

### CODE REVIEW

File: rdtools/test/irradiance_rescale_test.py [hunk 1]
```diff
@@ -7,19 +7,28 @@
...
-    expected = measured
-    assert_series_equal(rescaled, expected, check_exact=False)
+    if method == "error":
+        pytest.raises(
+            ValueError,
+            irradiance_rescale,
+            simple_irradiance,
+            simple_irradiance * 1.05,
+            method=method,
+        )
```
Comment (mdeceglie) [inline rdtools/test/irradiance_rescale_test.py:25]: Same question as above regarding use of `pytest.raises` without `with`

**Review by mdeceglie — COMMENTED** [review]
@martin-springer  Thanks for putting this together, a few questions but otherwise looks great.

**Comment by mdeceglie** [comment]
@martin-springer  is it possible to re-trigger the codecov report?

**Comment by martin-springer** [comment]
> @martin-springer is it possible to re-trigger the codecov report?

I do believe so, but haven't done that much myself yet. Looking through the quick start guide (https://docs.codecov.com/docs/quick-start), you can execute the upload of coverage reports from the command line. This page outlines the process (https://docs.codecov.com/docs/codecov-uploader). 

We might want to tweak the github action, to not only run when a new PR is created but also when commits are made to a PR. This should then take care of it. I can take a look at how we would need to change the config for that.

**Pull request merged**